thijsvanloef/palworld-server-docker
# Update server without delay if nobody is connected

## Context 

* The Updater should not wait to update the server, if there are no Players connected.

## Choices

* To check how many players are connected we are using RCON, since there is currently no other method
* If players connects after the script has sent the message, the player will not know that there will be an update (if no discord webhook is used). This reduces the chance of this happening.

## Test instructions
1. set the following environment variables: 
```env
AUTO_UPDATE_ENABLED: true
AUTO_UPDATE_WARN_MINUTES: 1
AUTO_UPDATE_CRON_EXPRESSION: "* * * * *"
```
2. Connect to the Server with a client
3.  change the buildid in `palworld\steamapps\appmanifest_2394010.acf` to a different value
4. verify that the script waits for the time set in `AUTO_UPDATE_WARN_MINUTES`
5. wait for the server to restart
6. change the buildid again without connecting
7. verify that the server is being stopped immediately after the cronjob was ran.

## Checklist before requesting a review
